src/main/java/com/epam/ta/reportportal/dao/LogRepositoryCustomImpl.java

```diff
@@ -131,4 +154,35 @@ public Integer getPageNumber(Long id, Filter filter, Pageable pageable) {
 
 		return BigDecimal.valueOf(rowNumber).divide(BigDecimal.valueOf(pageable.getPageSize()), RoundingMode.CEILING).intValue();
 	}
+
+	@Override
+	public boolean hasLogsAddedLately(Duration period, Long launchId, StatusEnum... statuses) {
+		List<JStatusEnum> jStatuses = Arrays.stream(statuses).map(it -> JStatusEnum.valueOf(it.name())).collect(toList());
+		Timestamp before = Timestamp.from(Instant.now().minusSeconds(period.getSeconds()));
```

Comment:
i can see similar logic of counting before timestamp in several methods. could you please make one common method. it exists in a previous version of rp. find it and implement something similar here

Reply:
done